### ISSUE TITLE
rush version should give error message when git username / email unset

### ISSUE THREAD

**Issue by townbully** [Microsoft/web-build-tools#738]
Currently, if running 'rush version' with out git username and/or email set, the command just fails with an opaque "ERROR: The command failed with exit code 1" message.  The email/name may be unset if running rush on a CI server, for example.  I needed to step through the rush code to determine what the problem was.  A more explicit error message would be very helpful here.

**Comment by nickpape-msft** [comment]
Good point, sounds like a bug. There is should be detection that occurs when you run other Rush commands (e.g. "rush install"). When you run "rush install" do you get a proper warning message or the "ERROR: The command failed with exit code 1"?

Also, are you using Git for your repo? [There are a few places where we may have inadvertantly added a Git dependency].

**Comment by townbully** [comment]
Hi, Nick:

Using 'rush install' doesn't produce this same error because I don't believe that function is making use of git.  The 'rush version' command does have this error because it uses git behind the scenes as part of its functionality and git fails to do things without the username/email set.

We are using git as our repo as part of a monorepo based project using rush.  It's okay if git is a dependency, just would be good to have error messages that more readily identify git as the source of a problem when git is unhappy.

**Comment by nickpape-msft** [comment]
> Using 'rush install' doesn't produce this same error because I don't believe that function is making use of git.

It does do a check for the e-mail address/name, but I noticed it only does that if you specify a `"gitPolicy"` in your `rush.json`, e.g.:

```json
  "gitPolicy": {
    "allowedEmailRegExps": [
      "[^@]+@contoso\\.com"
    ],
    "sampleEmail": "example@contoso.com"
  }
```

You might consider using that feature so developers or the CI don't accidentally get into a situation where they continually encounter the confusing error.

Meanwhile, we will definitely need to provide a better error message.

**Comment by townbully** [comment]
Thanks!  And thanks for the nice tool!

**Comment by nickpape-msft** [comment]
This should be fixed in master, you should see it in the 5.0.1 release!

**Comment by townbully** [comment]
Thanks again!

**Comment by HipsterZipster** [comment]
Thank you @townbully and @nickpape-msft for fixing this!
